addons/account/models/account_journal_dashboard.py

```diff
@@ -238,7 +239,8 @@ def _get_sale_purchase_graph_data(self):
         query_results = {r['journal_id']: r for r in self.env.cr.dictfetchall()}
         result = {}
         for journal in self:
-            currency = journal.currency_id or journal.company_id.currency_id
+            # User may have read access on the journal but not on the company
+            currency = journal.currency_id or self.env['res.currency'].browse(journal.company_id.sudo().currency_id.id)
```

Comment:
here it's not, because you have sometimes currency as a browse record and sometimes as an ID. Code will fail if it gets an ID, but it's already too much confusing if the code was flexible enough to accept both, in my opinion.

Reply:
I might not be fully aware maybe. Is there a type difference between `journal.currency_id` and `self.env['res.currency'].browse(id)`

Reply:
The idea is to only fetch the `id` of the currency with sudo and then use that `int` to get the record without sudo.

Reply:
yeah, but then you do `currency.round()` which don't work if currency is an ID instead of a browse record :shrug:

Reply:
Still don't see the problem ... To me it's always a record. Do I miss something?
```py
journal.currency_id  # a record (Many2One field on a model)
self.env['res.currency'].browse(journal.company_id.sudo().currency_id.id) # also a record
```

Reply:
oops, you're right. I totally misread that ^^

and it's a rather smart way of solving my oint, I must say :)